lib/internal/streams/pipeline.js

```diff
@@ -126,6 +126,10 @@ async function pump(iterable, writable, finish) {
   }
   let error;
   try {
+    if (writable.writableNeedDrain === true) {
+      await EE.once(writable, 'drain');
```

Comment:
This might never resolve if the stream is prematurely destroyed. I don't think this is a problem but would like a second opinion. @benjamingr @mcollina

Reply:
This whole PR/change is scary to me 😅

Reply:
I am not blocking because I don't have a good sense of how much of a breaking change this sort of change can be. It's possible it's not breaking and not a problem.

So I guess what I'm saying is "I don't know if this is a problem"

Reply:
I think it should be fine. Now that I think about it, it’s the same as inside the loop below.